### ISSUE TITLE
Wrong index variables used in loop nests

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#102]
## 🐛 Bug

Here's a generated loop nest:
 ```
for(size_t i101 = 0; i101 < T6.size[0]; ++i101 ) {
    for(size_t i102 = 0; i102 < T6.size[1]; ++i102 ) {
      T2[ ( i102 * T2.stride[0] ) + ( i103 * T2.stride[1] ) ]
         = T1[ ( i101 * T1.stride[0] ) + ( i102 * T1.stride[1] ) ];
    }
  }
```
Compilation fails as `i103` is undefined. 

The nest is generated from a modified softmax test. See https://github.com/naoyam/pytorch/commit/c38d89fdf0d178ef5e9d4bb988cf70e43ce45863#diff-538e381bc787011cc53c15caf54e126dR2770. It basically does the same arithmetic operations as  the original softmax test, but for a debugging purpose, all the parallelization-related tensor operations are removed. 

## To Reproduce

Run `testGPU_FusionSoftmax` modified in this commit: https://github.com/naoyam/pytorch/commit/c38d89fdf0d178ef5e9d4bb988cf70e43ce45863

**Comment by naoyam** [comment]
It seems this is a side effect of broadcasting into an input tensor. It's probably checked at `fusion.addInput`.
